### ISSUE TITLE
slices with higher code points can't write to utf8 streams

### ISSUE THREAD

**Issue by erg** [factor/factor#1611]
```factor
USE: sequences.extras
"å" sequence>slice "resource:a.txt" utf8 set-file-contents
```

I think this should work.

**Comment by mrjbq7** [comment]
This bug is really that ``slices`` aren't supported by streams.

**Comment by mrjbq7** [comment]
(I'm not sure they should be)

**Comment by mrjbq7** [comment]
Hmm, slices on byte-arrays work on binary streams.

So I guess maybe we should support slices on strings on character streams?

**Comment by erg** [comment]
I have to call ``>string io:write`` on the slice which seems dumb.

**Comment by mrjbq7** [comment]
You don't have to do that, you can just ``>string`` ... ``utf8 set-file-contents``...

But I see your issue, I just think there are a few reasons that we should make it only take strings instead of random sequences of random things that we hope are characters.

**Comment by mrjbq7** [comment]
Fixed in 0988df1e5a3bbb0a0612a84ef3f214cfc91078b3.

**Comment by erg** [comment]
Nice!

Another one:

``"a" sequence>slice "resource:a.txt" ascii set-file-contents``

**Comment by erg** [comment]
Also, we need some unit tests just to make sure this isn't a regression someday.

**Comment by mrjbq7** [comment]
what do you mean another one?  that doesn't work?

**Comment by erg** [comment]
Yeah, each encoding has its own bug!

**Comment by mrjbq7** [comment]
Oh i see, okay fixed in 6d417f4f71a9c38138edfdf39ba7686aa0595eae.
